src/core/email-client.js

```diff
@@ -53,20 +68,62 @@ function getFreshImap (cb) {
 }
 
 class EmailClient {
-    constructor () {
+    //constructor takes the full name of user as String
+    constructor (name) {
         var self = this;
+        self.name = name; 
 
         //  Get Imap obj and check emails
-        getFreshImap(function (err, imap) {
+        getFreshImap(function (err, imap, email, smtpConfig) {
             if (err) {
                 console.log(err);
             } else {
                 self.imap = imap;
+                self.email = email;
+                self.transporter = NodeMailer.createTransport(smtpConfig);
```

Comment:
name, imap, email, and transporter are all internal variables (correct me if I'm wrong) so add an underscore before their names, e.g. self._name, self._imap